aws/resource_aws_storagegateway_gateway_activation_key.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsStorageGatewayGatewayActivationKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsStorageGatewayGatewayActivationKeyCreate,
+		Read:   resourceAwsStorageGatewayGatewayActivationKeyRead,
+		Delete: schema.Noop,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"activation_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ip_address": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.SingleIP(),
+			},
+		},
+	}
+}
+
+func resourceAwsStorageGatewayGatewayActivationKeyCreate(d *schema.ResourceData, meta interface{}) error {
+	activationRegion := meta.(*AWSClient).region
+	ipAddress := d.Get("ip_address").(string)
+
+	err := resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		_, err := getStorageGatewayGatewayActivationKeyFromIP(activationRegion, ipAddress)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("error retrieving activation key from IP Address (%s): %s", ipAddress, err)
+	}
+
+	d.SetId(ipAddress)
+
+	return resourceAwsStorageGatewayGatewayActivationKeyRead(d, meta)
+}
+
+func resourceAwsStorageGatewayGatewayActivationKeyRead(d *schema.ResourceData, meta interface{}) error {
+	activationRegion := meta.(*AWSClient).region
+	activationKey, err := getStorageGatewayGatewayActivationKeyFromIP(activationRegion, d.Id())
+	if err != nil {
+		return fmt.Errorf("error reading activation key: %s", err.Err)
+	}
+
+	d.Set("activation_key", activationKey)
+	d.Set("ip_address", d.Id())
+
+	return nil
+}
+
+func getStorageGatewayGatewayActivationKeyFromIP(activationRegion, ipAddress string) (string, *resource.RetryError) {
+	client := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+		Timeout: time.Second * 10,
+	}
+	requestURL := fmt.Sprintf("http://%s/?activationRegion=%s", ipAddress, activationRegion)
```

Comment:
feels like this wants a "this is a special configuration and we don't support using resources in this manor; please use the AWS SDK where possible" disclaimer?